sdk/rust-examples/audio-event-triangulation/src/main.rs

```diff
@@ -0,0 +1,177 @@
+//! An implementation of triangulation of audio event data
+//!
+//! This demo takes in audio events in the form of a timestamp,
+//! GPS coordinates, and an analog rich audio sample, calculates the
+//! best effort triangulated location, returning the estimated location
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Copyright
+//!
+//! See the file `LICENSING.markdown` in the Veracruz root directory for licensing
+//! and copyright information.
+
+use anyhow;
+use std::{
+    convert::TryInto,
+    fs,
+    io,
+};
+use thiserror::Error;
+
+/// An individual audio event
+struct AudioEvent {
+    /// Timestamp of when even occured (currently unused)
+    #[allow(dead_code)]
+    timestamp: u32,
+    /// Location in 32-bit signed GPS coordinates
+    location: (i32, i32),
+    /// Samples in signed 16-bit PCM, bitrate is assumed to be
+    /// consistent for all samples
+    samples: Vec<i16>,
+}
+
+#[derive(Error, Debug)]
+pub enum AudioEventError {
+    #[error("Unable to decode audio event")]
+    InvalidAudioEvent,
+}
+
+impl AudioEvent {
+    /// Compute the power of the audio event, larger+closer audio
+    /// events result in higher power
+    fn power(&self) -> f64 {
+        // compute as avg per single sample, assumes common bitrate
+        let mut p: f64 = 0.0;
+        for x in self.samples.iter() {
+            p += x.abs() as f64;
+        }
+
+        p / (self.samples.len() as f64)
```

Comment:
What happens if `len()` is zero?

Reply:
That's a good question, I'll change this to return 0 in this case. A zero-length audio sample mapping to zero power makes sense.